tensorflow/contrib/ignite/python/ops/ignite_dataset_ops.py

```diff
@@ -0,0 +1,763 @@
+      return self.read_data(length).decode("utf-8")
+    if header == 101:
+      return None
+    raise Exception("Unknown binary type when expected string [type_id=%d]" \
+        % header)
+
+class IgniteDataset(Dataset):
+  """Apache Ignite is a memory-centric distributed database, caching, and
+     processing platform for transactional, analytical, and streaming workloads,
+     delivering in-memory speeds at petabyte scale. This contrib package
+     contains an integration between Apache Ignite and TensorFlow. The
+     integration is based on tf.data from TensorFlow side and Binary Client
+     Protocol from Apache Ignite side. It allows to use Apache Ignite as a
+     datasource for neural network training, inference and all other
+     computations supported by TensorFlow. Ignite Dataset is based on Apache
+     Ignite Binary Client Protocol.
+  """
+
+  def __init__(self, cache_name, host="localhost", port=10800, local=False,\
+    part=-1, page_size=100, username=None, password=None, certfile=None,\
+    keyfile=None, cert_password=None):
+    """Create a IgniteDataset.
+
+    Args:
+      cache_name: Cache name to be used as datasource.
+      host: Apache Ignite Thin Client host to be connected.
+      port: Apache Ignite Thin Client port to be connected.
+      local: Local flag that defines to query only local data.
+      part: Number of partitions to be queried.
+      page_size: Apache Ignite Thin Client page size.
+      username: Apache Ignite Thin Client authentication username.
+      password: Apache Ignite Thin Client authentication password.
+      certfile: File in PEM format containing the certificate as well as
+        any number of CA certificates needed to establish the certificate’s
+        authenticity.
+      keyfile: File containing the private key (otherwise the private key
+        will be taken from certfile as well).
+      cert_password: Password to be used if the private key is encrypted and a
+        password is necessary.
+    """
+    super(IgniteDataset, self).__init__()
+
+    with IgniteClient(host, port, username, password, certfile, keyfile,\
+        cert_password) as client:
+      client.handshake()
+      self.cache_type = client.get_cache_type(cache_name)
+
+    self.cache_name = ops.convert_to_tensor(cache_name, dtype=dtypes.string,\
+        name="cache_name")
+    self.host = ops.convert_to_tensor(host, dtype=dtypes.string, name="host")
+    self.port = ops.convert_to_tensor(port, dtype=dtypes.int32, name="port")
+    self.local = ops.convert_to_tensor(local, dtype=dtypes.bool, name="local")
+    self.part = ops.convert_to_tensor(part, dtype=dtypes.int32, name="part")
+    self.page_size = ops.convert_to_tensor(page_size, dtype=dtypes.int32,\
+        name="page_size")
+    self.username = ops.convert_to_tensor("" if username is None else username,\
+        dtype=dtypes.string, name="username")
+    self.password = ops.convert_to_tensor("" if password is None else password,\
```

Comment:
This (and `self.cert_password`) will encode potentially sensitive information in the `GraphDef` and send it over insecure channels. Is it necessary to include this information here?

Reply:
It's not necessary, but it's the simple way to start using.

We provide two ways to specify these parameters: via parameters of dataset or via environment variables on the nodes where dataset will be actually instantiated. As far as I understand, in the second case sensitive information won't be included into `GraphDef` and passed via insecure channels.

Reply:
Perhaps we should only support the environment variable, in that case? I'm concerned that novice users won't understand the distinction, take the simple route, and leak sensitive information.

However, security isn't my domain, so @martinwicke can you please make a determination here?